embedded/sql/stmt.go

```diff
@@ -2342,26 +2390,19 @@ func (i periodInstant) resolve(tx *SQLTx, params map[string]interface{}, asc, in
 }
 
 func (stmt *tableRef) referencedTable(tx *SQLTx) (*Table, error) {
-	var db *Database
-
-	if stmt.db != "" {
-		rdb, err := tx.catalog.GetDatabaseByName(stmt.db)
-		if err != nil {
-			return nil, err
-		}
-
-		db = rdb
+	if stmt.db != "" && stmt.db != tx.currentDB.name {
```

Comment:
The check for `if tx.currentDB == nil {` should be moved above this one.

Otherwise `tx.currentDB.name` will crash here if database is not selected.